nsv13/code/controllers/subsystem/starsystem.dm

```diff
@@ -0,0 +1,199 @@
+
+/datum/controller/subsystem/starsystem/proc/event_info() //Admin command for debugging output
+	var/datum/round_event_control/lone_hunter/LH = locate(/datum/round_event_control/lone_hunter) in SSevents.control
+	var/datum/round_event_control/belt_rats/BR = locate(/datum/round_event_control/belt_rats) in SSevents.control
+	message_admins("Current time: [world.time] | Last Combat: [last_combat_enter] | Modifier: [modifier] | [LH.name]: [LH.weight] | [BR.name]: [BR.weight]")
+
+/datum/controller/subsystem/starsystem/proc/weighting_reset() //All overmap combat events need to be populated in this proc
+	var/datum/round_event_control/lone_hunter/LH = locate(/datum/round_event_control/lone_hunter) in SSevents.control
+	var/datum/round_event_control/belt_rats/BR = locate(/datum/round_event_control/belt_rats) in SSevents.control
+	LH.weight = 0
+	BR.weight = 0
+
+/datum/controller/subsystem/starsystem/New()
+	. = ..()
+	instantiate_systems()
+
+/datum/controller/subsystem/starsystem/proc/instantiate_systems()
+	cycle_gameplay_loop() //Start the gameplay loop
+	cycle_bounty_timer() //Start the bounty timers
+	var/datum/starsystem/SS = new
+	hyperspace = SS //First system defaults to "hyperspace" AKA transit.
+	for(var/instance in subtypesof(/datum/starsystem))
+		var/datum/starsystem/S = new instance
+		systems += S
+
+///////SPAWN SYSTEM///////
+
+/datum/controller/subsystem/starsystem/proc/find_main_overmap()
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects)
+		if(OM.main_overmap)
+			return OM
+
+/datum/controller/subsystem/starsystem/proc/find_main_miner()
+	for(var/obj/structure/overmap/OM in GLOB.overmap_objects)
+		if(OM.main_miner)
+			return OM
+
+/datum/controller/subsystem/starsystem/proc/find_system(obj/structure/overmap/OM) //Used to determine what system a ship is currently in. Famously used to determine the starter system that you've put the ship in.
+	var/datum/starsystem/found
+	for(var/datum/starsystem/S in systems)
+		for(var/thez in SSmapping.levels_by_trait(S.level_trait))
+			if(thez == OM.z)
+				found = S
+				break
+	return found
+
+/datum/controller/subsystem/starsystem/proc/modular_spawn_enemies(obj/structure/overmap/OM, datum/starsystem/target_sys)//Select Ship to Spawn and Location via Z-Trait
+	var/list/levels = SSmapping.levels_by_trait(target_sys.level_trait)
+	if(levels?.len == 1)
+		var/turf/exit = get_turf(locate(round(world.maxx * 0.5, 1), round(world.maxy * 0.5, 1), levels[1])) //Plop them bang in the center of the system.
+		var/turf/destination = get_turf(pick(orange(20,exit)))
+		var/obj/structure/overmap/enemy = new OM(destination)
+		target_sys.add_enemy(enemy)
+	else if(levels?.len > 1)
+		message_admins("More than one level found for [target_sys]!")
+		return
+	else
+		message_admins("No levels forund for [target_sys]!")
+		return
+
+///////BOUNTIES//////
+
+/datum/controller/subsystem/starsystem/proc/bounty_payout()

```

Comment:
It would be better to have a component which sends a signal to something to process the payout.

Reply:
I think this is acceptable here, as it's just periodically paying things out. No need to overcomplicate things.

Reply:
This is intentionally a delayed process to avoid spamming the ship with messages every time a ship is destroyed/a sector is cleared. This delay also is intended to prevent clutch munitions orders mid-combat for more ordinance.